cosi/cosi_new.go

```diff
@@ -0,0 +1,130 @@
+package cosi
+
+import (
+	"crypto/cipher"
+
+	"github.com/dedis/kyber/abstract"
+)
+
+func Commit(suite abstract.Suite, s cipher.Stream) (abstract.Scalar, abstract.Point) {
+	// compute random scalar
+	// compute commitment
+	// return both
+}
+
+func AggregateCommitments(suite abstract.Suite, commitments []abstract.Point, mask *Mask) (abstract.Point, *Mask) {
```

Comment:
Same here, we want to aggregate the commitments along the way in the commit bottom-up phase of the CoSi communication tree. For that we need to be able to update the mask continuously until we reach the root node. As a consequence we also need to pass the mask from bottom to top I think.